File: Daily Test/Python/Last Alphabets 7.5.2025.py
s = input().strip().split()
d = {}
o = []

for w in s:
    k = w[-1]
    if k not in d:
        d[k] = []
        o.append(k)
    d[k].append(w)

for k in sorted(o):
    print(' '.join(d[k]))

File: Daily Test/Java/Last Alphabets 7.5.2025.java
import java.util.*;

public class WordGrouping {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] s = sc.nextLine().trim().split(" ");
        Map<Character, List<String>> map = new LinkedHashMap<>();

        for (String w : s) {
            char k = w.charAt(w.length() - 1);
            map.putIfAbsent(k, new ArrayList<>());
            map.get(k).add(w);
        }

        map.keySet().stream().sorted().forEach(key -> {
            System.out.println(String.join(" ", map.get(key)));
        });

        sc.close();
    }
}